internal/core/workers/runtime_watcher_worker/runtime_watcher_worker_test.go

```diff
@@ -0,0 +1,141 @@
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+// +build unit
+
+package runtime_watcher_worker
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	clockmock "github.com/topfreegames/maestro/internal/adapters/clock/mock"
+	instancemock "github.com/topfreegames/maestro/internal/adapters/instance_storage/mock"
+	pamock "github.com/topfreegames/maestro/internal/adapters/port_allocator/mock"
+	roomstoragemock "github.com/topfreegames/maestro/internal/adapters/room_storage/mock"
+	runtimemock "github.com/topfreegames/maestro/internal/adapters/runtime/mock"
+	"github.com/topfreegames/maestro/internal/core/entities"
+	"github.com/topfreegames/maestro/internal/core/entities/game_room"
+	porterrors "github.com/topfreegames/maestro/internal/core/ports/errors"
+	"github.com/topfreegames/maestro/internal/core/services/room_manager"
+	"github.com/topfreegames/maestro/internal/core/workers"
+)
+
+func TestProcessRuntimeEvents(t *testing.T) {
+	workerOptions := func(t *testing.T) (*gomock.Controller, *instancemock.MockGameRoomInstanceStorage, *runtimemock.MockRuntime, *workers.WorkerOptions) {
+		mockCtrl := gomock.NewController(t)
+
+		now := time.Now()
+		portAllocator := pamock.NewMockPortAllocator(mockCtrl)
+		roomStorage := roomstoragemock.NewMockRoomStorage(mockCtrl)
+		runtime := runtimemock.NewMockRuntime(mockCtrl)
+		instanceStorage := instancemock.NewMockGameRoomInstanceStorage(mockCtrl)
+		fakeClock := clockmock.NewFakeClock(now)
+		config := room_manager.RoomManagerConfig{RoomInitializationTimeoutMillis: time.Millisecond * 1000}
+		roomManager := room_manager.NewRoomManager(fakeClock, portAllocator, roomStorage, instanceStorage, runtime, config)
+
+		return mockCtrl, instanceStorage, runtime, &workers.WorkerOptions{
+			Runtime:     runtime,
+			RoomManager: roomManager,
+		}
+	}
+
+	t.Run("fails to start watcher", func(t *testing.T) {
```

Comment:
@gabrielcorado WDYT about adding one test case for when `w.roomManager.UpdateRoomInstance` call fails?

Reply:
Sure.